aspnetcore/web-api/advanced/formatting.md

```diff
@@ -168,80 +187,68 @@ public IActionResult Get()
 
 ::: moniker-end
 
-### Add XML format support
-
 ::: moniker range="<= aspnetcore-2.2"
 
-To add XML formatting support in ASP.NET Core 2.2 or earlier, install the [Microsoft.AspNetCore.Mvc.Formatters.Xml](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/) NuGet package.
+### Add XML format support
 
-::: moniker-end
+XML formatting requires the [Microsoft.AspNetCore.Mvc.Formatters.Xml](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/) NuGet package.
 
-XML formatters implemented using `System.Xml.Serialization.XmlSerializer` can be configured by calling <xref:Microsoft.Extensions.DependencyInjection.MvcXmlMvcBuilderExtensions.AddXmlSerializerFormatters*> in `Startup.ConfigureServices`:
+XML formatters implemented using <xref:System.Xml.Serialization.XmlSerializer> are configured by calling <xref:Microsoft.Extensions.DependencyInjection.MvcXmlMvcBuilderExtensions.AddXmlSerializerFormatters*>:
 
-[!code-csharp[](./formatting/sample/Startup.cs?name=snippet1&highlight=2)]
+[!code-csharp[](./formatting/sample/Startup.cs?name=snippet)]
 
-Alternatively, XML formatters implemented using `System.Runtime.Serialization.DataContractSerializer` can be configured by calling <xref:Microsoft.Extensions.DependencyInjection.MvcXmlMvcBuilderExtensions.AddXmlDataContractSerializerFormatters*> in `Startup.ConfigureServices`:
+The preceding code serializes results using `XmlSerializer`.
 
-```csharp
-services.AddMvc()
-    .AddXmlDataContractSerializerFormatters();
-```
+When using the preceding code, controller methods should return the appropriate format based on the request's `Accept` header.
```

Comment:
For output formatting, we query the XmlSerializers to determine if they support serializing the type. The XML serializers are more restrictive than the JSON ones, and they'll often refuse to serialize for various types (usually for things like not having a parameterless public ctor etc). The only way to diagnose this is by looking at MVC's logs. We get lots of questions about this since it's not obvious what went wrong. Can we add a blurb about this?